### PR TITLE
bsunanda:Run2-hcx78 Correct access of HcalTopology

### PR DESCRIPTION
**Pull request by bsunanda**
Also extend some of the services of HcalDDDRecConstant to enable finer indexing in eta/phi

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: CaloOnlineTools/HcalOnlineDb/src/WriteL1TriggerObjetsXml.cc [hunk 1]
```diff
@@ -37,24 +37,25 @@
 #include "CalibFormats/HcalObjects/interface/HcalDbRecord.h"
 #include "FWCore/Framework/interface/ESHandle.h"
 #include "Geometry/CaloTopology/interface/HcalTopology.h"
+#include "Geometry/HcalCommonData/interface/HcalDDDRecConstants.h"
 XERCES_CPP_NAMESPACE_USE 
 //
 // class decleration
 //
 
 class WriteL1TriggerObjetsXml : public edm::EDAnalyzer {
```
Comment (ianna) [inline CaloOnlineTools/HcalOnlineDb/src/WriteL1TriggerObjetsXml.cc]: @bsunanda - please, use thread safe analyser

File: Geometry/HcalCommonData/src/HcalDDDRecConstants.cc [hunk 2]
```diff
@@ -6,7 +6,7 @@
 #include "CLHEP/Units/GlobalPhysicalConstants.h"
 #include "CLHEP/Units/GlobalSystemOfUnits.h"
 
-//#define DebugLog
+#define DebugLog
```
Comment (ianna) [inline Geometry/HcalCommonData/src/HcalDDDRecConstants.cc]: @bsunanda - please, disable debug messages for code which goes into release.

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@ianna @mmusich @ggovi @mulhearn Please take a look and approve (?) this PR

**Comment by ianna** [comment]
+1

**Comment by bsunanda** [comment]
@mmusich @ggovi @mulhearn pinging again for approval. Want further improvement for HCAL geometry (used for BH of HGCAL) once this PR is integrated

**Comment by mmusich** [comment]
+1

**Comment by bsunanda** [comment]
@ggovi @mulhearn Could you approve this?

**Comment by bsunanda** [comment]
@ggovi @mulhearn @davidlange6 Can you push this PR into IB so that one can make the next step for topology usage?

**Comment by ggovi** [comment]
+1

**Pull request merged**